bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisBoxHandler.java

```diff
@@ -0,0 +1,371 @@
+            case "PROTOCOL":
+                properties.put("protocol", value);
+                break;
+            case "VERSION":
+                properties.put(PROPERTY_FIRMWARE_VERSION, value);
+                break;
+            case "RSSI":
+                properties.put("rssi", value);
+                break;
+            case "NAME":
+                properties.put("hostname", value);
+                break;
+        }
+    }
+
+    private void receivedUpdate(String function, String value) {
+        logger.trace("receivedUpdate(): {} {}", function, value);
+        switch (function) {
+            case "ONOFF":
+                updateState(CHANNEL_TYPE_POWER, OnOffType.from(value));
+                break;
+
+            case "SETPTEMP":
+                if (value.equals("32768")) {
+                    value = "0";
+                }
+                updateState(CHANNEL_TYPE_TARGETTEMP,
+                        new QuantityType<Temperature>(Double.valueOf(value) / 10.0d, SIUnits.CELSIUS));
+                break;
+            case "AMBTEMP":
+                if (Double.valueOf(value).isNaN()) {
+                    value = "0";
+                }
+                updateState(CHANNEL_TYPE_AMBIENTTEMP,
+                        new QuantityType<Temperature>(Double.valueOf(value) / 10.0d, SIUnits.CELSIUS));
+                break;
+            case "MODE":
+                updateState(CHANNEL_TYPE_MODE, new StringType(value));
+                break;
+            case "FANSP":
+                updateState(CHANNEL_TYPE_FANSPEED, new StringType(value));
+                break;
+            case "VANEUD":
+                updateState(CHANNEL_TYPE_VANESUD, new StringType(value));
+                break;
+            case "VANELR":
+                updateState(CHANNEL_TYPE_VANESLR, new StringType(value));
+                break;
+            case "ERRCODE":
+                properties.put("errorCode", value);
+                updateProperties(properties);
+                break;
+            case "ERRSTATUS":
+                properties.put("errorStatus", value);
+                updateProperties(properties);
+                if ("ERR".equals(value)) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "device reported an error");
+                }
+                break;
+        }
+    }
+
+    @SuppressWarnings("null")
+    private void handleMessage(String data) {
+        logger.trace("handleMessage(): Message received - {}", data);
+        if (data.equals("ACK") || data.equals("")) {
+            return;
+        }
+        if (data.startsWith(ID + ':')) {
+            synchronized (this) {
+                new IntesisBoxIdentity(data).value.forEach((name, value) -> receivedId(name, value));
+                return;
+            }
+        }
+        IntesisBoxMessage message = IntesisBoxMessage.parse(data);
+        switch (message.getCommand()) {
```

Comment:
`parse()` returns Nullable. You could declare its return value as NonNull or do a null check here.

Reply:
Done